server/services/transactionsTransformer.go

```diff
@@ -172,10 +172,15 @@ func (transformer *transactionsTransformer) rewardTxToRosettaTx(tx *transaction.
 }
 
 func (transformer *transactionsTransformer) normalTxToRosetta(tx *transaction.ApiTransactionResult) (*types.Transaction, error) {
-	hasValue := !isZeroAmount(tx.Value)
 	operations := make([]*types.Operation, 0)
 
-	if hasValue {
+	// Special handling of:
+	// - intra-shard contract calls, bearing value, which fail with signal error
+	// - direct contract deployments, bearing value, which fail with signal error
+	// For these, the protocol does not generate an explicit SCR with the value refund (before Sirius, in some cases, it did).
+	// However, since the value remains at the sender, we don't emit any operations in these circumstances.
+	transfersValue := isNonZeroAmount(tx.Value) && !transformer.featuresDetector.isContractDeploymentWithSignalErrorOrIntrashardContractCallWithSignalError(tx)
```

Comment:
Maybe this condition can be simplified (see PR description), but this will be postponed, to prevent any potential misinterpretation.